python/lsst/ts/m2gui/controltab/tab_rigid_body_pos.py

```diff
@@ -33,7 +37,326 @@ class TabRigidBodyPos(TabDefault):
         Table's title.
     model : `Model`
         Model class.
+
+    Attributes
+    ----------
+    model : `Model`
+        Model class.
     """
 
+    # The following two maximum values come from the ts_mtm2 that originated
+    # by vendor. Need to figure out the details in a latter time.
+
+    # Maximum movement in x, y, or z direction in millimeter
+    MAX_DISTANCE_IN_MM = 4
+
+    # Maximum rotation in rx, ry, or rz direction in microradian
+    MAX_ROTATION_IN_URAD = 20000
+
     def __init__(self, title, model):
         super().__init__(title, model)
+
+        # Relative target position
+        self._target_position_relative = self._set_target_position()
+
+        # Absolute target position
+        self._target_position_absolute = self._set_target_position()
+
+        # Buttons in this table
+        self._buttons = {
+            "save_position": set_button("Save Position", self._callback_save_position),
+            "go_to_home": set_button("Go to Home", self._callback_go_to_home),
+            "set_home": set_button("Set Home", self._callback_set_home),
+            "jog": set_button("Jog", self._callback_jog),
+            "clear_values_relative": set_button(
+                "Clear Values",
+                self._callback_clear_values,
+                self._target_position_relative,
+            ),
+            "go_to_position": set_button(
+                "Go to Position", self._callback_go_to_position
+            ),
+            "clear_values_absolute": set_button(
+                "Clear Values",
+                self._callback_clear_values,
+                self._target_position_absolute,
+            ),
+        }
+
+        # Rigid body position
+        self._position = self._set_position()
+
+        # Internal layout
+        self.widget().setLayout(self._create_layout())
+
+        self._set_signal_position(self.model.utility_monitor.signal_position)
+
+    def _set_target_position(self):
+        """Set the target position of rigid body.
+
+        Returns
+        -------
+        position : `dict`
+            Target position of the rigid body. The key is the axis. The value
+            is the `PySide2.QtWidgets.QDoubleSpinBox`.
+        """
+
+        num_digit_after_decimal = (
+            self.model.utility_monitor.NUM_DIGIT_AFTER_DECIMAL_DISPLACEMENT
+        )
+
+        position = dict()
+        for axis in self._get_axes():
+            double_spin_box = QDoubleSpinBox()
+            double_spin_box.setDecimals(num_digit_after_decimal)
+            double_spin_box.setSingleStep(get_tol(num_digit_after_decimal))
+
+            if axis.startswith("r"):
+                double_spin_box.setRange(
```

Comment:
LSST preference/requirement is to use deg (or derived units, arcmin & arcsec) in SAL (see e.g. MTMount XML or various discussions over time in Slack). Please either change this to show in degrees (or even better in arcsec), which are more natural to think of for human operators. Or if still stick with urad (which you shall probably convert for JSON messages; although LSST rules don't apply for those messages, as you have to somehow ship the values to EFD anyway and that has to be in degrees, it's better to use units which are preferred), please use Unicode for micro - either type it directly or as https://www.compart.com/en/unicode/U+00B5 e.g. \u00B5.